src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -25,6 +25,18 @@ export function ThemeProvider({
     () => buildCustomProperties(themeConfig, unstableGlobalTheming),
     [unstableGlobalTheming, themeConfig],
   );
+  const backgroundColor = customProperties['--p-surface-background'];
+  const color = customProperties['--p-text-on-surface'];
+
+  useEffect(() => {
+    if (unstableGlobalTheming) {
+      document.body.style.backgroundColor = backgroundColor;
+      document.body.style.color = color;
+    } else {
+      document.body.style.backgroundColor = null;
```

Comment:
If we're not using global theming then will `customProperties['--p-surface-background']` (and text-on-surface) be set? If not then do we this logic toggle at all?

When unstableGlobalTheming changes, then customProperties will change and the keys mentioned will either be a particular color (if unstableGlobalTheming is true)  or undefined (if unstableGlobalTheming is false)

```
useEffect(() => {
  document.body.style.backgroundColor = backgroundColor;
  document.body.style.color = color;
}, [backgroundColor, color])
```

Reply:
`'--p-surface-background'` in this case is a key lookup that returns the actual value, so yep it works. It sets the style attribute inline on body.

That’s why we need the `unstableGlobalTheming` dependency, so if that value changes we need to set those inline styles to null (remove them) so they fallback to the body styles defined in global.scss.

Reply:
I think the missing part here is that setting `document.body.style.backgroundColor = undefined` is a no-op and does not remove the inline value, whereas `document.body.style.backgroundColor = null` does remove the value.

I updated the code to be a bit simpler.

Reply:
Thats the gotcha I was missing! the `|| null` works a treat